org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/CorePlugin.java

```diff
@@ -187,7 +195,30 @@ private UserNotification createUserNotification() {
         return new ConsoleUserNotification();
     }
 
+    /*
+     * https://resheim.net/2008/10/connecting-through-proxy.html
+     */
+    private IProxyService retrieveProxyService() {
+        final String CORE_NET_BUNDLE = "org.eclipse.core.net";
+        Bundle bundle = Platform.getBundle(CORE_NET_BUNDLE);
+        while (bundle.getState() != Bundle.ACTIVE) {
+            try {
+                Thread.sleep(1000);
+            } catch (InterruptedException e) {
+                e.printStackTrace();
```

Comment:
We shouldn't wait for the a bundle to become active. If a dependency exists from plug-in A to B, then B will be always active before A. If it's not the case, than further investigation is needed.